### ISSUE TITLE
"Please enter an address to continue" at checkout even though user has entered an address (1016)

### ISSUE THREAD

**Issue by siliconforks** [woocommerce/woocommerce-paypal-payments#1025]
### Describe the Bug

The error message "Please enter an address to continue" appears on the checkout page even though the user has entered an address.

### To Reproduce

1. The WooCommerce PayPal Payments plugin should be fully configured with credit cards enabled.
2. In the WordPress admin section, under WooCommerce -> Settings -> General, "Default customer location" should be set to "No location by default".
3. In the WordPress admin section, under WooCommerce -> Settings -> Accounts & Privacy, "Allow customers to place orders without an account" should be checked.
4. Make sure shipping is enabled and working (i.e., make sure you have a shipping zone and shipping method configured).
5. As a user who is not logged in (I recommend you clear all cookies to be sure), add an item to your cart and go to the checkout page.
6. Enter a valid billing address, leave the "Ship to a different address?" checkbox unchecked, and select the payment option provided by WooCommerce PayPal Payments.
7. Click the "Debit or Credit Card" button.

### Screenshots

![checkout](https://user-images.githubusercontent.com/1156399/204015115-667c6019-15af-4f62-b6c7-2165445062ce.png)

### Expected Behavior

The above steps should work without any error message.

### Actual Behavior

The error message "Please enter an address to continue" is displayed even though you entered an address.

Curiously, if you reload the checkout page, then the issue disappears.  The "Debit or Credit Card" button will work.

### Environment

 - WordPress Version: 6.1.1
 - WooCommerce Version: 7.1.0
 - Plugin Version: 2.0.0
 - Browser [e.g. Chrome, Safari] and Version: tested with multiple browsers
 - Any other plugins installed: no

### Additional Details

There are several reports of this issue in the support forum:

https://wordpress.org/support/topic/please-enter-address-to-continue/
https://wordpress.org/support/topic/plugin-errors-29/

<details><summary>System status</summary>

  ```
  `
### WordPress Environment ###

WordPress address (URL): https://example.test/woocommerce-paypal-payments
Site address (URL): https://example.test/woocommerce-paypal-payments
WC Version: 7.1.0
REST API Version: ✔ 7.1.0
WC Blocks Version: ✔ 8.7.5
Action Scheduler Version: ✔ 3.4.0
Log Directory Writable: ✔
WP Version: 6.1.1
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: Apache/2.4.54 (Win64) OpenSSL/1.1.1p PHP/7.4.30
PHP Version: 7.4.30
PHP Post Max Size: 100 MB
PHP Time Limit: 120
PHP Max Input Vars: 1000
cURL Version: 7.70.0
OpenSSL/1.1.1p

SUHOSIN Installed: –
MySQL Version: 5.7.39-log
Max Upload Size: 100 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 7.1.0
WC Database Prefix: wp_
Total Database Size: 3.85MB
Database Data Size: 2.38MB
Database Index Size: 1.47MB
wp_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_actions: Data: 0.02MB + Index: 0.11MB + Engine InnoDB
wp_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_comments: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_options: Data: 1.48MB + Index: 0.06MB + Engine InnoDB
wp_postmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_posts: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_usermeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_admin_notes: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_admin_note_actions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_product_lookup: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_stats: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_product_attributes_lookup: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_download_directories: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_wc_rate_limits: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB

### Post Type Counts ###

attachment: 1
page: 7
post: 2
product: 1

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (2) ###

WooCommerce PayPal Payments: by WooCommerce – 2.0.0
WooCommerce: by Automattic – 7.1.0

### Inactive Plugins (2) ###

Akismet Anti-Spam: by Automattic – 5.0.1
Hello Dolly: by Matt Mullenweg – 1.7.2

### Settings ###

API Enabled: –
Force SSL: –
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –
Enforce Approved Product Download Directories: ✔

### WC Pages ###

Shop base: #6 - /?page_id=6
Cart: #7 - /?page_id=7
Checkout: #8 - /?page_id=8
My account: #9 - /?page_id=9
Terms and conditions: ❌ Page not set

### Theme ###

Name: Storefront
Version: 4.2.0
Author URL: https://woocommerce.com/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –

### WooCommerce PayPal Payments ###

Onboarded: ✔
Shop country code: US
WooCommerce currency supported: ✔
Advanced Card Processing available in country: ✔
Pay Later messaging available in country: ✔
Webhook status: –
Vault enabled: ✔
Logging enabled: –
Reference Transactions: ✔
Used PayPal Checkout plugin: –
Tracking enabled: ✔

### Admin ###

Enabled Features: activity-panels
analytics
coupons
customer-effort-score-tracks
experimental-products-task
experimental-import-products-task
experimental-fashion-sample-products
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
multichannel-marketing
mobile-app-banner
navigation
onboarding
onboarding-tasks
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
shipping-label-banner
subscriptions
store-alerts
transient-notices
woo-mobile-welcome
wc-pay-promotion
wc-pay-welcome-page

Disabled Features: minified-js
new-product-management-experience
settings

Daily Cron: ✔ Next scheduled: 2022-11-26 14:30:16 +00:00
Options: ✔
Notes: 36
Onboarding: skipped

### Action Scheduler ###

Complete: 5
Oldest: 2022-11-25 14:31:37 +0000
Newest: 2022-11-25 14:31:38 +0000

Pending: 1
Oldest: 2022-11-26 14:30:54 +0000
Newest: 2022-11-26 14:30:54 +0000


### Status report information ###

Generated at: 2022-11-25 15:26:30 +00:00
`
  ```

</details>

**Comment by InpsydeNiklas** [comment]
Thanks for the report! The best workaround for now is likely to configure `Default customer location` with `Shop country/region` instead of `No location by default`, as the latter can also cause other issues under certain circumstances (e.g. when the billing country field is hidden).

**Comment by Sirvijver** [comment]
Hello @siliconforks 

The fix for this bug was included in last release (2.0.1), please test that this update resolved the issue for you and open a new thread if this didn't resolve it for you.

Kind regards, 
Joost
